src/Illuminate/Database/Connectors/ConnectionFactory.php

```diff
@@ -96,6 +97,58 @@ protected function createReadPdo(array $config)
      */
     protected function createPdoResolver(array $config)
     {
+        if (array_key_exists('host', $config)) {
+            return $this->createPdoResolverWithHosts($config);
+        }
+
+        return $this->createPdoResolverWithoutHosts($config);
+    }
+
+    /**
+     * Create a new Closure that resolves to a PDO instance with a specific host or an array of hosts.
+     *
+     * @param  array  $config
+     * @return \Closure
+     */
+    protected function createPdoResolverWithHosts(array $config)
+    {
+        return function () use ($config) {
+            if (! is_array($config['host'])) {
+                $hosts = [$config['host']];
+            } else {
+                $hosts = $config['host'];
+                shuffle($hosts);
+            }
+
+            $lastHost = end($hosts);
+            foreach ($hosts as $host) {
+                $config['host'] = $host;
+
+                try {
+                    return $this->createConnector($config)->connect($config);
+                } catch (PDOException $e) {
+	                if ($host !== $lastHost) {
+	                    $this->container->make('Illuminate\Contracts\Debug\ExceptionHandler')->report($e);
+                    }
+                }
+            }
+
+            if (empty($hosts)) {
+                throw new InvalidArgumentException('Database hosts array cannot be empty');
+            }
+
+            throw $e;
```

Comment:
The only way for the code to reach that throw statement is if all hosts have failed. If so, the `$e` variable will be set from the catch block in the last iteration of the foreach loop.
If any host succeeded, the method would have returned already.